### DISCUSSION TITLE
Header Data Format not accept the dataclass

### DISCUSSION THREAD

**Discussion by jialiang-yin** [fastapi/fastapi#6704] [Questions] [answered]
I have a question about the Header accepted data format. 
As the following code shows, I want to send a Header, parse the content of this Header and GET image streamer from my server(localhost). 
But if I entered some string to the Header, that will show me the error: 

```
Error: Unprocessable EntityResponse 
bodyDownload{   "detail": [     {       "loc": [         "header",         "image-option"       ],       
"msg": "instance of ImageOption, tuple or dict expected",       
"type": "type_error.dataclass",       
"ctx": {         "class_name": "ImageOption"       }     }   ] }
```

 This is only a test script. Laterly I will let the ImageOption inherit other class. But the premier test script won't let me do this. Or the Header can only be some str, int or float. extra.

My OS is ArchLinux, and x86_64. I think, the OS is not that problem. 

```import asyncio
import io
from dataclasses import dataclass
from time import sleep
from typing import Optional, Dict

import uvicorn
from fastapi import FastAPI, Header, Query, Path
from starlette.responses import StreamingResponse
import cv2
from harvesters.core import Harvester

from autorein.websocket_util.ws_image import Image, create_test_image

app = FastAPI()


@dataclass
class ImageOption:
    data: Optional[str] = None


def capcure_image(cam_source):
    cap = cv2.VideoCapture(cam_source)
    ret, image = cap.read()
    return image


def encode_img_to_bytes():
    img = capcure_image(0)
    ret, img_encoded = cv2.imencode('.png', img)
    return img_encoded.tobytes()

async def video_streamer():
    while True:
        cv2img_encode = encode_img_to_bytes()
        await asyncio.sleep(0.01)
        yield (b'--frame\r\n' b'Content-Type: image/png\r\n\r\n' +
               bytearray(cv2img_encode) + b'\r\n')

@app.get("/vector_image")
async def video_feed(image_option: ImageOption = Header(...)):
    print(image_option)
    return StreamingResponse(video_streamer(), media_type="multipart/x-mixed-replace; boundary=frame")


if __name__ == '__main__':
    uvicorn.run('fastapi_test:app', host='127.0.0.1', port=8000, reload=True)```

**Comment by Mause** [comment]
What are you expecting to happen here? Headers are strings, not objects

**Comment by jialiang-yin** [comment]
I will add still some integers and strings in that ImageOption class. 
The server reads the Options from the ImageOption and then return some images. 

Shortly, set some options, parse them and according to the options, the images will be sent.
Setting options better with the object injection.

**Comment by jialiang-yin** [comment, marked as answer]
I figure it out, that the Depends keyword can be used for this case. 
Change this line
`async def video_feed(image_option: ImageOption = Header(...)):`
 to below
`async def video_feed(image_option: ImageOption = Depends(ImageOption)):`
